unity/Assets/Scripts/Unity/UnityMap.cs

```diff
@@ -157,5 +167,23 @@ public List<GameObject> GetAffectedTiles(Coordinate coordinate)
             }
             return new List<GameObject>();
         }
+
+        IEnumerator PlaceTileDelayed(ITile tile, GameObject gameObject, Material material, Vector3 coordinate, Quaternion quaternion, EType type)
```

Comment:
Can we make this private? What's the IEnumerator return type for?

Reply:
Ok, you need an IEnumerator for the Coroutines, I see. All good then.

Reply:
Ok, last point: Don't think you actually need line 175. The tile sets it's own material anyway.

That means you can get rid of a bunch of parameters for the method, namely all but 'tile' and 'type'.